colcon_bundle/verb/__init__.py

```diff
@@ -53,3 +53,36 @@ def check_and_mark_bundle_tool(bundle_base, *, this_build_tool='colcon'):
         os.makedirs(bundle_base, exist_ok=True)
 
     marker_path.write_text(this_build_tool + '\n')
+
+
+def check_and_mark_bundle_version(bundle_base, *, this_bundle_version):
+    """
+    Check the bundle version marker file for the previous bundle version.
+
+    The marker filename is `.bundle_version`.
+
+    :param str bundle_base: The bundle directory
+    :param str this_bundle_version: The version of the bundle to be built
+    :raises RuntimeError: if the bundle version does not match the passed
+    in bundle version
+    """
+    marker_path = Path(bundle_base) / '.bundle_version'
+    if marker_path.parent.is_dir():
+        previous_bundle_version = 1
+        if marker_path.is_file():
+            previous_bundle_version = int(marker_path.read_text().rstrip())
+        if previous_bundle_version == this_bundle_version:
+            marker_path.write_text(str(this_bundle_version) + '\n')
+            return
+        raise RuntimeError(
+            'The bundle directory "{bundle_base}" was used to create a '
+            'version {previous_bundle_version} bundle. Please remove the '
+            'bundle directory or pick a different one. If you would prefer '
+            'to use the old version use the --bundle-version argument. '
+            'Bundle version 2 has multiple improvements to bundling speed'
+            'and other optimizations, it is highly '
+            'recommended.'.format_map(locals()))
+    else:
+        os.makedirs(bundle_base, exist_ok=True)
+
+    marker_path.write_text(str(this_bundle_version) + '\n')
```

Comment:
sounds like marker_path should be opened r/w at the beginning of the function instead of doing this every time.